Scripts/dea_bandindices.py

```diff
@@ -0,0 +1,255 @@
+                  'BAI': lambda ds: (1.0 / ((0.10 - ds.red) ** 2 +
+                                            (0.06 - ds.nir) ** 2)),
+
+                  # Normalised Difference Built-Up Index, Zha 2003
+                  'NDBI': lambda ds: (ds.swir1 - ds.nir) /
+                                     (ds.swir1 + ds.nir),
+
+                  # Normalised Difference Snow Index, Hall 1995
+                  'NDSI': lambda ds: (ds.green - ds.swir1) /
+                                     (ds.green + ds.swir1),
+
+                  # Normalised Difference Water Index, McFeeters 1996
+                  'NDWI': lambda ds: (ds.green - ds.nir) /
+                                     (ds.green + ds.nir),
+
+                  # Modified Normalised Difference Water Index, Xu 2006
+                  'MNDWI': lambda ds: (ds.green - ds.swir1) /
+                                      (ds.green + ds.swir1),
+
+                  # Automated Water Extraction Index (no shadows), Feyisa 2014
+                  'AWEI_ns': lambda ds: (4 * (ds.green - ds.swir1) -
+                                        (2.5 * ds.nir * + 2.75 * ds.swir2)),
+
+                  # Automated Water Extraction Index (shadows), Feyisa 2014
+                  'AWEI_sh': lambda ds: (ds.blue + 2.5 * ds.green -
+                                         1.5 * (ds.nir + ds.swir1) -
+                                         2.5 * ds.swir2),
+
+                  # Water Index, Fisher 2016
+                  'WI': lambda ds: (1.7204 + 171 * ds.green + 3 * ds.red -
+                                    70 * ds.nir - 45 * ds.swir1 -
+                                    71 * ds.swir2),
+
+                  # Tasseled Cap Wetness, Crist 1985
+                  'TCW': lambda ds: (0.0315 * ds.blue + 0.2021 * ds.green +
+                                     0.3102 * ds.red + 0.1594 * ds.nir +
+                                    -0.6806 * ds.swir1 + -0.6109 * ds.swir2),
+
+                  # Tasseled Cap Greeness, Crist 1985
+                  'TCG': lambda ds: (-0.1603 * ds.blue + -0.2819 * ds.green +
+                                     -0.4934 * ds.red + 0.7940 * ds.nir +
+                                     -0.0002 * ds.swir1 + -0.1446 * ds.swir2),
+
+                  # Tasseled Cap Brightness, Crist 1985
+                  'TCB': lambda ds: (0.2043 * ds.blue + 0.4158 * ds.green +
+                                     0.5524 * ds.red + 0.5741 * ds.nir +
+                                     0.3124 * ds.swir1 + -0.2303 * ds.swir2),
+
+                  # Clay Minerals Ratio, Drury 1987
+                  'CMR': lambda ds: (ds.swir1 / ds.swir2),
+
+                  # Ferrous Minerals Ratio, Segal 1982
+                  'FMR': lambda ds: (ds.swir1 / ds.nir),
+
+                  # Iron Oxide Ratio, Segal 1982
+                  'IOR': lambda ds: (ds.red / ds.blue)
+    }
+
+    try:
+
+        # Select a water index function based on 'water_index'
+        index_func = index_dict[index]
```

Comment:
Rather than catching `KeyError` in this large block of code it is best to use a more explicit error handling like:

```python
index_func = index_dict.get(index)
if index_func is None:
    raise ValueError("..whatever error message...")
```

That way only errors due to user input will be reported as such.

Similarly with `AttributeError` it's best to check explicitly if data variable with the supplied name exists in the dataset rather than catching any `AttributeError` which might come from your own typos rather than user input.